dts/bindings/radio_fem/skyworks,sky66112-11.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+description: >
+    This is a representation of the Skyworks SKY66112-11 Radio Front-End module.
+
+compatible: "skyworks,sky66112-11"
+
+include: base.yaml
+
+properties:
+    ctx-gpios:
+        type: phandle-array
+        required: false
+        description: >
+            Gpio of the SOC controlling CTX pin of the SKY66112-11
+
+    crx-gpios:
+        type: phandle-array
+        required: false
+        description: >
+            Gpio of the SOC controlling CRX pin of the SKY66112-11
+
+    ctx-settle-time-us:
+        type: int
+        default: 23
+        description: >
+            Settling time in microseconds from activation of CTX to transmit.
+
+            The default value has been determined experimentally: it is the
+            minimum time interval between asserting the CTX signal and starting
+            the radio transmission, which does not generate unacceptable
+            interference level on adjacent channels. The tests concerned a
+            configuration in which nRF5x SoC controlling the SKY66112-11.
+            The value can be adjusted for other tuned configurations.
+
+    crx-settle-time-us:
+        type: int
+        default: 5
+        description: >
+            Settling time in microseconds from activation of CRX to receive.
+
+            The default value has been determined experimentally: it is the
+            minimum time interval between asserting the CRX signal and starting
+            the radio reception, which does not degrade transmission quality
+            indicators (compared to a constantly turned on FEM LNA). The tests
+            concerned a configuration in which nRF5x SoC controlling
+            the SKY66112-11. The value can be adjusted for other tuned
+            configurations.
```

Comment:
I'm a bit uncomfortable with the justifications for these defaults (for ctx-settle-time-us and crx-settle-time-us). It seems like they might not be following the rules, which are laid out here:

https://docs.zephyrproject.org/latest/guides/dts/bindings.html

When it says that it's been tuned for a particular configuration and may need to be adjusted for other configurations, it seems like they are potentially not independent, which would break this rule:

> If changing one property would require changing another to create a
> consistent configuration then those properties should be made
> required.

It also seems like it's just tuned for one SoC, which in general is not appropriate for defaults -- the consensus is that those should be made required, and recommended defaults for different situations should be documented.

Reply:
> it seems like they are potentially not independent

I think they are independent. I am aware of users that tuned only TX timing, because RX was correct for their application.


> It also seems like it's just tuned for one SoC

That's a valid point. It looks these parameters work correctly for the whole SoC family (at least nRF52). I'm just thinking, that we are adding this binding to sdk-nrf, which is intended only for nRF SoCs. Perhaps we could relax this rule and keep SoCs related default in binding to make creating DTSs easier for SDK users?

If not, I'm fine with making these parameters required and documenting recommended values.

Reply:
> I think they are independent. I am aware of users that tuned only TX timing, because RX was correct for their application.

OK, that resolves this concern. Thanks for confirming.

> I'm just thinking, that we are adding this binding to sdk-nrf, which is intended only for nRF SoCs. Perhaps we could relax this rule and keep SoCs related default in binding to make creating DTSs easier for SDK users?

Yeah, I thought about that too. The downsides of this approach are:

- If we ever upstream this binding, it is unlikely to be accepted with these defaults. So at that point, the defaults would probably switch to `required` and existing user applications would break. Might not be a real problem if that never happens.

- In general, I have observed that when NCS behaves inconsistently with zephyr, it causes significant user confusion. Upstream vs. downstream flash partitioning and multi-image builds are the main example here. So things like this generally make me nervous. Might not be a real problem here.

What about this compromise? The driver can use DT_NODE_HAS_PROP() to check if the relevant properties are defined.

- If they are defined, just use them from the DT using DT_PROP() (or the INST versions of those macros).

- If they are not defined, the driver emits a warning and uses the current defaults in the binding.

If that's not acceptable, I would defer to area maintainers on whether making an exception in this case is worth it.

Reply:
I think we should decide now how to handle that this driver is generic, what is discussed under gain comment. I'll get to this later.